docs/modules/development/pages/rest/rest-api.adoc

```diff
@@ -1,121 +1,201 @@
+If the IP is valid and the node is not valid, because the node is an intermediate resource, a 400 will be returned.
+* If something not expected is received from the Service or DAO Layer when processing any HTTP request, 500 (INTERNAL_SERVER_ERROR) is returned.
+* Any problem related to incoming parameters (for example, validations) generates 400 (BAD_REQUEST).
 
-== Identifying Resources
+== Identifying resources
 
 Some endpoints deal in resources, which are identified by resource IDs.
-Since every resource is ultimately parented under some node, identifying the node that contains a resource is the first step in constructing a resource ID.
+Since every resource is ultimately parented under a node, identifying the parent node is the first step in constructing a resource ID.
 Two styles are available for identifying the node in a resource ID:
 
-[options="header, %autowidth"]
+[options="header" "autowidth", cols="1,2"]
 |===
-| Style            | Description                                                                           | Example
-| `node[ID]`       | Identifies a node by its database ID, which is always an integer                      | `node[42]`
-| `node[FS:FID]`   | Identifies a node by its foreign-source name and foreign-ID, joined by a single colon | `node[Servers:115da833-0957-4471-b496-a731928c27dd]`
+| Style
+| Description
+| Example
+
+| `node[ID]`
+| Identifies a node by its database ID, which is always an integer.
+| `node[42]`
+
+| `node[FS:FID]`
+| Identifies a node by its `foreign-source` name and `foreign-ID`, joined by a single colon.
+| `node[Servers:115da833-0957-4471-b496-a731928c27dd]`
 |===
 
-The node identifier is followed by a period, then a resource-type name and instance name.
+The node identifier is followed by a period, then a resource-type name and an instance name.
 The instance name's characteristics may vary from one resource-type to the next.
 A few examples:
 
-[options="header, %autowidth"]
+[options="header" "autowidth", cols="1,2"]
 |===
-| Value                              | Description
-| `nodeSnmp[]`                       | Node-level (scalar) performance data for the node in question.
-                                       This type is the only one where the instance identifier is empty.
-| `interfaceSnmp[eth0-04013f75f101]` | A layer two interface as represented by a row in the SNMP `ifTable`.
-                                       The instance identifier is composed of the interface's `ifName` and its `ifPhysAddress` (if it has one).
-| `dskIndex[_root_fs]`               | The root filesystem of a node running the Net-SNMP management agent.
+| Value
+| Description
+
+| `nodeSnmp[]`
+| Node-level (scalar) performance data for the node in question.
+This type is the only one where the instance identifier is empty.
+
+| `interfaceSnmp[eth0-04013f75f101]`
+| A layer two interface as represented by a row in the SNMP `ifTable`.
```

Comment:
```suggestion
| A layer 2 interface as represented by a row in the SNMP `ifTable`.
```
One of the times we should use the numeral instead of word. :-)